### PR TITLE
Error message: Getting head of empty path.

### PR DESCRIPTION
**Pull request by MartinNowak**
Create a file with name "appTest.d" and this content:

```
/+ dub.json:
{
	"name": "appTest",
}
+/

void main(){}
```

Execute dub with command: dub appTest.d
Error is shown: Getting head of empty path.

The issue is the use of upper case letters in the name. The issue disappears by changing the name to apptest:

```
/+ dub.json:
{
	"name": "apptest",
}
+/
```

### CODE REVIEW

**Comment by MartinNowak** [comment]
I fiddled around a bit with giving the recipe a fake path (e.g. `path.d-dub.sdl`), but it was too much churn for such a minor issue.

**Comment by s-ludwig** [comment]
It may make sense to add a field `PackageFormat format` to `BuildSettingsTemplate` that stores the format of the source and use that for the decision instead of the path. Should have a very small impact on the source and still produces the right error message for single-file packages.

**Comment by MartinNowak** [comment]
> It may make sense to add a field PackageFormat format to BuildSettingsTemplate that stores the format of the source and use that for the decision instead of the path.

Done, I was a bit hesitant to do this, at some point we might support more formats or even make this pluggable, but chances for extensibility aren't too big.

File: source/dub/recipe/packagerecipe.d [hunk 1]
```diff
@@ -106,6 +107,11 @@ struct SubPackage
 	PackageRecipe recipe;
 }
 
+enum RecipeFormat
+{
+	JSON,
+	SDL,
+}
```
Comment (s-ludwig) [inline source/dub/recipe/packagerecipe.d]: There is already a `PackageFormat` at package_.d:35, that one should be moved to here. I wouldn't mind renaming it to `RecipeFormat` (with a compatibility alias in package_.d), though, as the current name doesn't fit so well anymore (it predates `PackageRecipe).

**Comment by MartinNowak** [comment]
Done @s-ludwig

**Comment by s-ludwig** [comment]
This should just need a rebase, so that the dmd-beta failure goes away.

**Comment by Geod24** [comment]
This does not seem to trigger in master and can be closed.

**Comment by andre2007** [comment]
I can confirm, issue is reproducible on master.

**Pull request closed without merging**